### PR TITLE
[P2T support No.86、XXX] paddle.[max、min、mean、sum、prod]

### PR DESCRIPTION
**Pull request by cangtianhuang**
### 1. [paddle.max](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/max_cn.html#max)

查询 [torch.max 与 paddle.max 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.max.html) ，发现是 `参数用法不一致`，为其编写 mapping 转换配置后，注册 ReduceRule 类

paddle 的 axis 参数支持多维规约，而 torch.max 仅支持单一维度，因此直接将对应的 api 更换为 torch.amax，可实现多维规约

paddle 与 torch 的参数用法不一致，需要预处理：axis 若为列表或元组，需遍历元素，提取 0D Tensor的值，最后转换为 tuple；axis 若为 0D Tensor，直接提取值；axis 若为 1D Tensor，使用 tolist 转为 list 再转为 tuple

### 2. [paddle.min](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/min_cn.html#min)

查询 [torch.min 与 paddle.min 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.min.html) ，发现是 `参数用法不一致`，为其编写 mapping 转换配置后，注册 ReduceRule 类

与 paddle.min 类似，对应的 api 更换为 torch.amin，其他操作一致

### 3. [paddle.mean](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/mean_cn.html#mean)

查询 [torch.mean 与 paddle.mean 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.mean.html) ，发现是 `参数用法不一致`，为其编写 mapping 转换配置后，注册 ReduceRule 类

torch.mean 支持多维规约，但不支持 dim 为 None 时 keepdim 为 True，因此判断 axis 是否为 None，进行分支处理，最后再根据 keepdim 恢复相应维度

### 4. [paddle.sum](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/api/paddle/sum_cn.html#sum)

查询 [torch.sum 与 paddle.sum 对照表](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/guides/model_convert/convert_from_pytorch/api_difference/torch/torch.sum.html) ，发现是 `参数用法不一致`，为其编写 mapping 转换配置后，注册 ReduceRule 类

torch.sum 支持多维规约，预处理后直接调用

## 其他修改

paddle.prod 与上述四个 api 均属于 reduce 相关 api，因此检查先前的实现 ProdRule，优化逻辑后，合并至 ReduceRule 中，更新其 mapping 转换配置

## 测试结果

paddle.max、paddle.min、paddle.mean、paddle.sum 的全部配置均通过 accuracy 测试，配置移入 support2torch_1；paddle.prod 的所有 support2torch 配置也复测通过：

image

### CODE REVIEW

**Review by Cutelemon6 — APPROVED** [review]
LGTM
